api/envoy/extensions/rbac/matchers/upstream_ip_port/v3/upstream_ip_port_matcher.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package envoy.extensions.rbac.matchers.upstream_ip_port.v3;
+
+import "envoy/config/core/v3/address.proto";
+import "envoy/type/v3/range.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.rbac.matchers.upstream_ip_port.v3";
+option java_outer_classname = "UpstreamIpPortMatcherProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: RBAC upstream IP matcher plugin]
+// [#extension: envoy.rbac.matchers.upstream_ip_port]
+
+// This is configuration for matching upstream ip (and port if provided). Port will only be matched
+// if IP match is successful.
+message UpstreamIpPortMatcher {
+  // A CIDR block that will be used to match the upstream IP.
+  // This matcher requires a filter in the chain to have saved the upstream address in the
+  // filter state before the matcher is executed by RBAC filter. The state should be saved with key
+  // `envoy.stream.upstream_address` (See
+  // :repo:`upstream_address.h<source/common/stream_info/upstream_address.h>`).
+  // Also, See :repo:`proxy_filter.cc<
+  // source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc>` for an example of a
+  // filter which populates the FilterState.
+  // Both Ipv4 and Ipv6 ranges can be matched.
+  config.core.v3.CidrRange upstream_ip = 1 [(validate.rules).message = {required: true}];
+
+  // A port range that will be used to match the upstream port.
```

Comment:
Nit: point out this is optional?

Reply:
Done! Thanks @htuch

Reply:
I don't see this done? Can you indicate on the field docs that if no port range is supplied any port will match?